src/Agent.Listener/Agent.cs

```diff
@@ -35,10 +34,16 @@ public override void Initialize(IHostContext hostContext)
 
         public async Task<int> ExecuteCommand(CommandSettings command)
         {
+            AssemblyLoadContext loadContext = AssemblyLoadContext.GetLoadContext(typeof(Agent).GetTypeInfo().Assembly);
             try
             {
                 _inConfigStage = true;
+                _completedCommand.Reset();
                 _term.CancelKeyPress += CtrlCHandler;
+
+                //register a SIGTERM handler
+                loadContext.Unloading += Agent_Unloading;
```

Comment:
Did you say this is just for Linux (not OSX)? If this doesn't work for all platforms, can you leave a comment in the source code to clarify.